src/net/sourceforge/kolmafia/persistence/HolidayDatabase.java

```diff
@@ -820,65 +727,94 @@ public String toString() {
-    return getHoliday(dateTime, false);
+  public static void setHoliday(final String holiday) {
+    HOLIDAY_OVERRIDE = holiday.isBlank() ? null : holiday;
   }
 
   /** Returns the KoL holiday associated with the given date in the real world. */
-  public static String getHoliday(final ZonedDateTime dateTime, final boolean showPrediction) {
+  public static List<String> getHolidays(final ZonedDateTime dateTime) {
+    var holidays = new ArrayList<String>();
+
     String gameHoliday = getGameHoliday(dateTime);
+    if (gameHoliday != null) holidays.add(gameHoliday);
+
     String realHoliday = getRealLifeHoliday(dateTime);
+    if (realHoliday != null) holidays.add(realHoliday);
 
-    if (showPrediction && realHoliday == null) {
-      if (gameHoliday != null) {
-        return gameHoliday + " today";
-      }
+    if (holidays.contains("St. Sneaky Pete's Day") && holidays.contains("Feast of Boris")) {
+      holidays.clear();
+      holidays.add("Drunksgiving");
+    }
 
-      int calendarDay = getDayInKoLYear(dateTime);
-      gameHoliday = getGameHoliday((calendarDay + 1) % 96);
+    if (HOLIDAY_OVERRIDE != null) holidays.add(HOLIDAY_OVERRIDE);
 
-      if (gameHoliday != null) {
-        return gameHoliday + " tomorrow";
-      }
+    return holidays;
+  }
 
-      for (int i = 2; i < 96; ++i) {
-        gameHoliday = HolidayDatabase.getGameHoliday((calendarDay + i) % 96);
+  public static String getHoliday() {
+    return getHoliday(DateTimeManager.getRolloverDateTime());
+  }
 
-        if (gameHoliday != null) {
-          return HolidayDatabase.getDayCountAsString(i) + " until " + gameHoliday;
-        }
-      }
+  public static String getHoliday(final ZonedDateTime dateTime) {
+    return String.join(" / ", getHolidays(dateTime));
+  }
 
-      return "";
-    }
+  public static List<String> getEvents() {
+    return getEvents(DateTimeManager.getRolloverDateTime());
+  }
+
+  public static List<String> getEvents(final ZonedDateTime dateTime) {
+    var list = new ArrayList<>(getHolidays(dateTime));
 
-    if (gameHoliday == null && realHoliday == null) {
-      return "";
+    // Include this pseudo-event because the day before Labor Day essentially has +10 rollover
+    // adventures
+    if ("Lab&oacute;r Day".equals(getHolidayInDays(dateTime, 1))) {
+      list.add("Lab&oacute;r Day Eve");
     }
 
-    if (gameHoliday == null) {
-      return realHoliday;
+    var statDay = getStatDay(dateTime);
+    if (statDay != Stat.NONE) {
+      list.add(statDay.toString() + " Day");
     }
 
-    if (realHoliday == null) {
-      return gameHoliday;
+    return list;
+  }
+
+  public static String getHolidaySummary() {
+    return getHolidaySummary(DateTimeManager.getRolloverDateTime());
+  }
+
+  public static String getHolidaySummary(final ZonedDateTime dateTime) {
+    String realHoliday = getRealLifeHoliday(dateTime);
+
+    if (realHoliday != null) return getHoliday();
```

Comment:
probably want to pass in the `dateTime` here.

Also you're calculating the real holiday twice but whatever.